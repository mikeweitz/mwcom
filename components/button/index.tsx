import cx from 'classnames';

export interface ButtonProps
    extends React.ButtonHTMLAttributes<HTMLButtonElement> {
    children: React.ReactNode;
    className?: string;
    key?: string | number;
}

import styles from './styles.module.scss';

export default function Button({ children, className, ...rest }: ButtonProps) {
    return (
        <button className={cx(styles.button, className)} {...rest}>
            {children}
        </button>
    );
}
