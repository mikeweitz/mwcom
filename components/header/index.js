import * as S from './styled-elements';
import { Container } from '../../styles/grid';

import { copy } from '../../data';

const Header = () => {
  const {
    header: { title, email, github, linkedin }
  } = copy;
  return (
    <S.Heading id="header">
      <Container>
        <S.Group>
          <S.Title>{title}</S.Title>
        </S.Group>

        <address>
          <S.StyledLink href="#">
            <S.LinkSpan>
              {email.address}
              {email.domain}
            </S.LinkSpan>
          </S.StyledLink>
          <S.StyledLink href={github.url}>
            <S.LinkSpan>{github.text}</S.LinkSpan>
          </S.StyledLink>
          <S.StyledLink href={linkedin.url}>
            <S.LinkSpan>{linkedin.text}</S.LinkSpan>
          </S.StyledLink>
        </address>
      </Container>
    </S.Heading>
  );
};

export default Header;
