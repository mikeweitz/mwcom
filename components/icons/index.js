const Print = ({fill, alt}) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
    >
      {alt && <title>{alt}</title>}
      <path d="M19 8H5c-1.66 0-3 1.34-3 3v6h4v4h12v-4h4v-6c0-1.66-1.34-3-3-3zm-3 11H8v-5h8v5zm3-7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-1-9H6v4h12V3z" fill={fill || '#fff'}/>
      <path d="M0 0h24v24H0z" fill="none" />
    </svg>
  );
};

const Arrow = ({ fill, alt, style }) => {  
  return (
    <svg 
      xmlns="http://www.w3.org/2000/svg" 
      data-tags="remove-circle, cancel, close, remove, delete"
      viewBox="0 0 1200 1200"       
      >
        {alt && <title>{alt}</title>}
        <path fill={fill || '#fff'} d="M599.992 0L131.243 703.131H383.79V1200h432.422V703.131h252.546L599.992 0z"/>
    </svg>
  )
}

const Close = ({ fill, alt }) => {
  return (
    <svg 
    xmlns="http://www.w3.org/2000/svg" 
    viewBox="0 0 96 96">
      {alt && <title>{alt}</title>}
      <polygon fill={ fill || '#fff' } points="96,14 82,0 48,34 14,0 0,14 34,48 0,82 14,96 48,62 82,96 96,82 62,48 "/>
    </svg>
  );
}

export { Print, Arrow, Close };


