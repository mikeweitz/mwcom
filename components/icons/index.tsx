type IconProps = {
    fill: string;
    alt?: string;
    style?: any;
    width?: string | number;
    height?: string | number;
};

const Print = ({ fill, alt, ...rest }: IconProps) => {
    return (
        <svg
            xmlns="https://www.w3.org/2000/svg"
            width="24"
            height="24"
            viewBox="0 0 24 24"
            {...rest}
        >
            {alt && <title>{alt}</title>}
            <path
                d="M19 8H5c-1.66 0-3 1.34-3 3v6h4v4h12v-4h4v-6c0-1.66-1.34-3-3-3zm-3 11H8v-5h8v5zm3-7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-1-9H6v4h12V3z"
                fill={fill || '#fff'}
            />
            <path d="M0 0h24v24H0z" fill="none" />
        </svg>
    );
};

const Arrow = ({ fill, alt, ...rest }: IconProps) => {
    return (
        <svg
            xmlns="https://www.w3.org/2000/svg"
            data-tags="remove-circle, cancel, close, remove, delete"
            viewBox="0 0 1200 1200"
            {...rest}
        >
            {alt && <title>{alt}</title>}
            <path
                fill={fill || '#fff'}
                d="M599.992 0L131.243 703.131H383.79V1200h432.422V703.131h252.546L599.992 0z"
            />
        </svg>
    );
};

const Close = ({ fill, alt, ...rest }: IconProps) => {
    return (
        <svg {...rest} xmlns="https://www.w3.org/2000/svg" viewBox="0 0 96 96">
            {alt && <title>{alt}</title>}
            <polygon
                fill={fill || '#fff'}
                points="96,14 82,0 48,34 14,0 0,14 34,48 0,82 14,96 48,62 82,96 96,82 62,48 "
            />
        </svg>
    );
};

const Play = ({ fill, alt, ...rest }: IconProps) => {
    return (
        <svg
            className="MuiSvgIcon-root jss80 MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
            aria-hidden="true"
            {...(alt && { title: alt })}
            {...rest}
        >
            {alt && <title>{alt}</title>}
            <path
                fill={fill || '#fff'}
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 14.5v-9l6 4.5-6 4.5z"
            ></path>
        </svg>
    );
};
export { Print, Arrow, Close, Play };
