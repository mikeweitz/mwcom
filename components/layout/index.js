import React, { useEffect, useLayoutEffect, useState } from 'react';
import { useInterval } from '../../hooks/useInterval';
import { initGA, logPageView } from '../../util/analytics';
import Header from '../header';
import Footer from '../footer';
import * as S from './styled-elements';

const Layout = ({ children }) => {
  const useIsomorphicEffect =
    typeof window !== 'undefined' ? useLayoutEffect : useEffect;

  const delay = 8000;

  const [hueRotation, setHueRotation] = useState(null);

  const changeHueRotation = () => {
    const hue = Math.round(Math.random() * -180);
    console.log('Change bg hue', hue);
    setHueRotation(hue);
  };

  useInterval(() => {
    changeHueRotation();
  }, delay);

  useIsomorphicEffect(() => {
    if (window && !window.GA_INITIALIZED) {
      initGA();
      window.GA_INITIALIZED = true;
    }
    logPageView();
    return () => {
      console.log('unmount');
    };
  }, []);

  return (
    <S.Main $hue={hueRotation} $delay={delay}>
      <S.ContentWrap>
        <Header />
        {children}
        <Footer />
      </S.ContentWrap>
    </S.Main>
  );
};
export default Layout;
