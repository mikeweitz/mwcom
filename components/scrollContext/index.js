import React, { useState, useEffect, useLayoutEffect } from 'react';
import { throttle } from 'lodash';
const isBrowser = typeof window !== `undefined`;

const useIsomorphicEffect = isBrowser ? useLayoutEffect : useEffect;

const getScroll = () => ({
  x: isBrowser ? window.scrollX : 0,
  y: isBrowser ? window.scrollY : 0,
  isScrolled: false,
});

const ScrollContext = React.createContext();

function useScrollContext() {
  const context = React.useContext(ScrollContext);
  if (context === undefined) {
    throw new Error('useScrollContext must be used within a ScrollProvider');
  }
  return context;
}

const ScrollProvider = ({ children, scrollThreshold = 50 }) => {
  const [scroll, setScroll] = useState({
    x: 0,
    y: 0,
    isScrolled: false,
    scrolling: false,
  });

  useIsomorphicEffect(() => {
    if (!isBrowser) {
      return;
    }
    let timer = null;
    const delay = 250;
    const handleScroll = throttle(
      () => {
        clearTimeout(timer);
        const coords = getScroll();
        const newScroll = {
          ...coords,
          isScrolled: Math.abs(coords.y) > scrollThreshold,
          scrolling: true,
        };
        setScroll({
          ...newScroll,
        });
        timer = setTimeout(() => {
          setScroll({ ...newScroll, scrolling: false });
        }, delay + 10);
      },
      delay,
      {
        leading: true,
      }
    );

    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, []);

  return (
    <ScrollContext.Provider value={scroll}>{children}</ScrollContext.Provider>
  );
};

export { ScrollProvider, useScrollContext };
