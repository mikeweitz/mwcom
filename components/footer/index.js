import React, { Component } from 'react';

import * as S from './styled-elements';
import { animateScroll } from 'react-scroll';
import { Container } from '../../styles/grid';
import { Print } from '../icons';
import Link from 'next/link'

class Footer extends Component {
  constructor() {
    super();
  }

  state = {
    hoverPrint: false,
  };

  onPrint = e => {
    return window.open('/static/pdf/michael-weitzman-resume.pdf', 'download');
  };

  handleToTop = e => {
    animateScroll.scrollToTop();
  };

  handleMouseEnter = () => {
    this.setState({ hoverPrint: true });
  };

  handleMouseLeave = () => {
    this.setState({ hoverPrint: false });
  };
  render() {
    const { hoverPrint } = this.state;
    return (
      <S.Footer id="footer">
        <Container>
          <S.ButtonPrint
            onMouseEnter={this.handleMouseEnter}
            onMouseLeave={this.handleMouseLeave}
            onClick={this.onPrint}
          >
            <S.PrintIcon $hover={hoverPrint}>
              <Print />
            </S.PrintIcon>
            Print this
          </S.ButtonPrint>

          <S.ButtonTop onClick={this.handleToTop}>Back to top</S.ButtonTop>
        </Container>
        <Container>
          <Link href="/playlists">Monthly Playlists</Link>
        </Container>
      </S.Footer>
    );
  }
}

export default Footer;
