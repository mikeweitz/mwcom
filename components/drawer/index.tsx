import cx from 'classnames';
import { SetStateAction, useEffect, useRef } from 'react';

import { useScrollContext } from '@mw/components/scrollContext';
import { Close } from '@mw/components/icons';

import styles from './styles.module.scss';

interface DrawerProps {
    active: boolean;
    handleClose: React.Dispatch<SetStateAction<null | Record<string, any>>>;
    children?: React.ReactNode;
}

export default function Drawer({ active, children, handleClose }: DrawerProps) {
    const scroll = useScrollContext();
    const scrollRef = useRef(null);

    const onClick = () => {
        handleClose(null);
    };

    useEffect(() => {
        if (scrollRef && scrollRef.current) {
            scrollRef.current.scrollTop = 0;
        }
    }, [active]);

    return (
        <>
            <div
                className={cx(styles.drawer, {
                    [styles.scrolled]: scroll.isScrolled,
                    [styles.scrolling]: scroll.scrolling,
                    [styles.active]: active,
                })}
            >
                <section
                    className={cx(styles.section, {
                        [styles.active]: active,
                        [styles.scrolled]: scroll.isScrolled,
                    })}
                >
                    <div className={styles['scroll-container']} ref={scrollRef}>
                        <article>{children}</article>
                    </div>
                </section>
                <button
                    className={cx(styles.close, {
                        [styles.active]: active,
                    })}
                    onClick={onClick}
                    // onMouseEnter={() => setHoverClose(true)}
                    // onMouseLeave={() => setHoverClose(false)}
                >
                    <div className={styles.icon}>
                        <Close
                            alt="Close"
                            width="50%"
                            height="auto"
                            fill={'#ffffff'}
                        />
                    </div>
                </button>
            </div>
            {active && <div className={styles.screen}></div>}
        </>
    );
}
