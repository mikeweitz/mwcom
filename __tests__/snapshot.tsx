import { render } from '@testing-library/react';
import Header from '../components/header/index';
import { ScrollProvider } from '@mw/components/scrollContext';

it('renders page header unchanged', () => {
    const { container } = render(
        <ScrollProvider>
            <Header />
        </ScrollProvider>
    );
    expect(container).toMatchSnapshot();
});
