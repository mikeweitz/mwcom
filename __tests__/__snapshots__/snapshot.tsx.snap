// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders page header unchanged 1`] = `
<div>
  <header
    class="header"
    id="header"
  >
    <div
      class="overflow"
    >
      <button
        class="menu-button"
      >
        <span
          class="menu-dot menu-dot-one"
        />
        <span
          class="menu-dot menu-dot-two"
        />
        <span
          class="menu-dot menu-dot-three"
        />
      </button>
      <div
        class="page-top"
      >
        <h2
          class="title"
        >
          <a
            class="title-link"
            href="/"
          >
            Michael Weitzman
          </a>
        </h2>
        <address
          class="nav-wrap"
        >
          <a
            class="link"
            href="#"
          >
            <span
              class="link-span"
            >
              weitzly@gmail.com
            </span>
          </a>
          <a
            class="link"
            href="https://github.com/mikeweitz"
          >
            <span
              class="link-span"
            >
              github
            </span>
          </a>
          <a
            class="link"
            href="https://www.linkedin.com/in/mikeweitzman/"
          >
            <span
              class="link-span"
            >
              linkedin
            </span>
          </a>
          <a
            class="link"
            href="/playlists"
          >
            <span
              class="link-span"
            >
              playlists
            </span>
          </a>
        </address>
      </div>
      <div
        class="page-scrolled"
      >
        <h2
          class="title title-small"
        >
          <a
            class="title-link"
            href="/"
          >
            <span
              class="initials"
            >
              MW
            </span>
          </a>
        </h2>
        <address
          class="nav-wrap"
        >
          <a
            class="link"
            href="#"
          >
            <span
              class="link-span"
            >
              weitzly@gmail.com
            </span>
          </a>
          <a
            class="link"
            href="https://github.com/mikeweitz"
          >
            <span
              class="link-span"
            >
              github
            </span>
          </a>
          <a
            class="link"
            href="https://www.linkedin.com/in/mikeweitzman/"
          >
            <span
              class="link-span"
            >
              linkedin
            </span>
          </a>
          <a
            class="link"
            href="/playlists"
          >
            <span
              class="link-span"
            >
              playlists
            </span>
          </a>
        </address>
      </div>
      <a
        href="/"
      >
        <div
          class="logo"
        >
          <img
            class="img"
            src="/logo.png"
          />
        </div>
      </a>
    </div>
  </header>
</div>
`;
